packages/core/src/lib/context/createContext.ts

```diff
@@ -75,14 +75,19 @@ export function makeCreateContext({
       return result.data as any;
     };
 
-    async function withRequest(req: IncomingMessage, res?: ServerResponse) {
-      contextToReturn.req = req;
-      contextToReturn.res = res;
-      if (!config.session) {
-        return contextToReturn;
+    async function withRequest(newReq: IncomingMessage, newRes?: ServerResponse) {
+      const newContext = createContext({
+        session,
+        sudo,
+        req: newReq,
+        res: newRes,
+      });
+
+      if (config.session) {
+        newContext.session = await config.session.get({ context: newContext });
```

Comment:
a note here: I considered doing this instead:
```ts
        const newSession = await config.session.get({ context: newContext });
        return createContext({
          session: newSession,
          sudo,
          req: newReq,
          res: newRes,
        });
```

however, that would end up calling createContext twice. Also, the `session` property in the createContext initial arguments isn't used by any of the "modification" functions (sudo, exitSudo, withSession), so it wouldn't end up having the same issue that the req/res objects did initially.